src/main/java/hudson/plugins/promoted_builds/PromotionTargetAction.java

```diff
@@ -26,12 +26,10 @@ public PromotionTargetAction(AbstractBuild<?,?> build) {
     }
 
     public AbstractBuild<?,?> resolve(PromotionProcess parent) {
-        AbstractProject<?,?> j = parent.getOwner();
-        if (j==null)    return null;
-        return j.getBuildByNumber(number);
+    	return resolve();
```

Comment:
I'm not sure what was the idea of the original method. Probably the idea was to make it static.
I'll check the code